### PR TITLE
refactor(core): Refactor cli command tests (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
extracted out of #9696

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included

### CODE REVIEW

File: packages/cli/src/commands/worker.ts [hunk 1]
```diff
@@ -498,7 +498,7 @@ export class Worker extends BaseCommand {
 		}
 
 		// Make sure that the process does not close
-		await new Promise(() => {});
+		if (!inTest) await new Promise(() => {});
```
Comment (ivov) [inline packages/cli/src/commands/worker.ts:501]: I take it there's no other choice but in general we should avoid sprinkling `inTest` checks in production code.

File: packages/cli/test/integration/shared/utils/testCommand.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	beforeAll(async () => {
+		await testDb.init();
+	});
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	afterAll(async () => {
+		await testDb.terminate();
```
Comment (ivov) [inline packages/cli/test/integration/shared/utils/testCommand.ts:24]: Nit - Some `testDb` calls happen in the suite, some happen here. This dries up tests but might condition us to e.g. forget about `testDb.terminate` in tests that are not about CLI commands and do not have this convenience? Personally I'd prefer not to switch back and forth to see what the suite is doing. Really minor though.

**Review by ivov — APPROVED** [review]
So much cleaner, thank you!

File: packages/cli/src/commands/worker.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/src/commands/worker.ts:501]: This was "temporary" code that [Jan had to add](https://github.com/n8n-io/n8n/pull/5461), after I broke this in #5452 .
At some point I hope to fix this properly, until then we need this ugly check 😞

**Comment by janober** [comment]
Got released with [`n8n@1.47.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.47.0)

**Comment by janober** [comment]
Got released with [`n8n@1.47.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.47.0)

**Pull request merged**